pep-0690.rst

```diff
@@ -209,38 +403,73 @@ This PEP proposes to add a new ``importlib.eager_imports()`` context manager,
 so the above technique can be less verbose and doesn't require comments to
 clarify its intent::
 
+    from importlib import eager_imports
+
     with eager_imports():
         import foo
         import bar
 
 Since imports within context managers are always eager, the ``eager_imports()``
 context manager can just be an alias to a null context manager. The context
-manager does not force all imports to be recursively eager: ``foo`` and ``bar``
-will be imported eagerly, but imports within those modules will still follow
-the usual laziness rules.
+manager's effect is not transitive: ``foo`` and ``bar`` will be imported
+eagerly, but imports within those modules will still follow the usual laziness
+rules.
 
 The more difficult case can occur if an import in third-party code that can't
-easily be modified must be forced to be eager. For this purpose, we propose to
-add an API to ``importlib`` that can be called early in the process to specify
-a list of module names within which all imports will be eager::
+easily be modified must be forced to be eager. For this purpose,
+``importlib.enable_lazy_imports()`` takes an optional keyword-only ``eager``
+argument, which can be set to a container of module names within which all
+imports will be eager::
 
-    from importlib import set_eager_imports
+    from importlib import enable_lazy_imports
 
-    set_eager_imports(["one.mod", "another"])
+    enable_lazy_imports(eager=["one.mod", "another"])
 
 The effect of this is also shallow: all imports within ``one.mod`` will be
 eager, but not imports in all modules imported by ``one.mod``.
 
-``set_eager_imports()`` can also take a callback which receives a module name and returns
-whether imports within this module should be eager::
+The ``eager`` parameter of ``set_eager_imports()`` can also be set to a callback
+which receives a module name and returns whether imports within this module
+should be eager::
 
     import re
-    from importlib import set_eager_imports
-    
+    from importlib import enable_lazy_imports
+
     def eager_imports(name):
         return re.match(r"foo\.[^.]+\.logger", name)
 
-    set_eager_imports(eager_imports)
+    enable_lazy_imports(eager=eager_imports)
+
+If Python was executed with the ``-L`` flag, then lazy imports will already be
+globally enabled, and the only effect of calling ``enable_lazy_imports()`` will
+be to globally set the eager module names/callback.
+
+``enable_lazy_imports()`` may be called more than once, with subsequent calls
+having only the effect of globally replacing the eager module names/callback.
```

Comment:
So, if I called things in this order:

```py
enable_lazy_imports(excluding=['foo', 'bar'])
enable_lazy_imports()
```

then it erases "foo" and "bar" as eager imports and makes all imports lazy again?  The PEP should be explicit about this.

Reply:
I think the updated version I'm about to push is clear on this; let me know if you think it's still unclear.